src/turing_model.jl

```diff
@@ -0,0 +1,226 @@
+
+Example for a robust model: `@formula(y, ...), data; priors=CustomPrior(Normal(0, 2.5), Normal(10, 5), Exponential(1))`
+
+# `standardize`
+
+Whether `true` or `false` to standardize your data to mean 0 and standard deviation 1
+before inference. Some science fields prefer to analyze and report effects in terms of
+standard devations. Also, whenever measurement scales differs, it is often suggested to
+standardize the effects for better comparison. By default, `turing_model` sets `standardize=false`.
+"""
+function turing_model(
+    formula::FormulaTerm, data::D; family="normal", priors=DefaultPrior(), standardize=false
+) where {D}
+
+    # extract y, X and Z
+    y = data_response(formula, data)
+    X = data_fixed_effects(formula, data)
+    Z = data_random_effects(formula, data)
+
+    # μ and σ identities
+    μ_X = 0
+    σ_X = 1
+    μ_y = 0
+    σ_y = 1
+
+    if standardize
+        μ_X, σ_X, X = standardize_predictors(X)
+        μ_y, σ_y, y = standardize_predictors(y)
+    end
+
+    # Random-Effects Conditionals
+    if has_ranef(formula)
+        # Likelihood Conditionals
+        # TODO
+    else
+        # Likelihood Conditionals
+        if family == "normal"
+            if priors isa DefaultPrior
+                custom_prior = CustomPrior(
+                    TDist(3), LocationScale(median(y), mad(y), TDist(3)), nothing
+                )
+            else
+                custom_prior = priors
+            end
+            @model function normal_model(
+                y,
+                X;
+                predictors=size(X, 2),
+                μ_X=μ_X,
+                σ_X=σ_X,
+                prior=custom_prior,
+                residual=1 / std(y),
+            )
+                α ~ prior.intercept
+                β ~ filldist(prior.predictors, predictors)
+                σ ~ Exponential(residual)
+                y ~ MvNormal(α .+ X * β, σ^2 * I)
+                return (; α, β, σ, y)
+            end
+            return normal_model(y, X)
+        elseif family == "student"
+            if priors isa DefaultPrior
+                custom_prior = CustomPrior(
+                    TDist(3), LocationScale(median(y), mad(y), TDist(3)), Gamma(2, 0.1)
+                )
+            else
+                custom_prior = priors
+            end
+            @model function student_model(
+                y,
+                X;
+                predictors=size(X, 2),
+                μ_X=μ_X,
+                σ_X=σ_X,
+                prior=custom_prior,
+                residual=1 / std(y),
+            )
+                α ~ prior.intercept
+                β ~ filldist(prior.predictors, predictors)
+                σ ~ Exponential(residual)
+                ν ~ prior.auxiliary
+                y ~ arraydist(LocationScale.(α .+ X * β, σ, TDist.(ν)))
+                return (; α, β, σ, ν, y)
+            end
```

Comment:
Same as above. With more than 120 lines, this method is too long to be easily readable